### DISCUSSION TITLE
[FR]: How to use js Formula？

### DISCUSSION THREAD

**Discussion by ChristinaHyh** [RafaelGB/obsidian-db-folder#584] [Q&A] [answered]
### Contact Details

hyahuime@gmail.com

### Present your request

I have set up to enable js formulas, however, It doesn't seem to work. Can you give some examples? Thank you.
I try to use formulas to realize the function of counting down to the deadline of Project. But I don't know how to get today's date, I used js function written by myself, but it can't work. 

### For which platform do you request this request??

Desktop

**Comment by RafaelGB** [comment, marked as answer]
hi @ChristinaHyh ,

Could you share your js function?

you also can check our documentation

**Reply by RafaelGB** [reply]
https://rafaelgb.github.io/obsidian-db-folder/features/Formulas/

**Reply by ChristinaHyh** [reply]
Thank you for your reply. I have read the documents, but I still don't know how to make it work.
My js funtion is as follows:
`function getToday() {
    var Date_today=new Date();
    var year=Date_today.getFullYear();
    var month=Date_today.getMonth()+1;
    var date=Date_today.getDate();
    if (month >= 1 && month <= 9) {
        month = "0" + month;
    }
    if (date >= 0 && date <= 9) {
        date = "0" + date;
    }
    var TodayDate = year + "-" + month + "-" + date;
    return TodayDate;
}`
And I used `${db.js.getToday();}` in the Formula input. 
<img width="397" alt="2022-11-07_195408" src="https://user-images.githubusercontent.com/48902684/200305262-774523bc-c370-4d12-861e-f9b209ee752d.png">

<img width="597" alt="2022-11-07_194149" src="https://user-images.githubusercontent.com/48902684/200304619-fb34eec3-becc-4a98-a9e4-7f615b9dc85b.png">

I don't know why it can't work, did I set the wrong filename? Or use the wrong method?
Thank you

**Reply by RafaelGB** [reply]
we need a full example in our documentation. Will be added with next versions.

the format of the formula :
`myfunction.js`

```js
function myfunction(args) {
  // Your logic
}

//required to add your function to the plugin
module.exports = myfunction
```

then you can use it

`${db.js.myfunction(...)}`

**Reply by ChristinaHyh** [reply]
Thank you~
I still have some questions, in this example, I need to name the js file as myfunction.js, right? And this js file should be included in the js folder that set in the DB folder plugin config, am I right?
Should this (`module.exports = myfunction`) be included in myfunction.js file or in obsidian plugin config?
Thank you for your help.

**Reply by RafaelGB** [reply]
you need to include it inside of your js

So if you have a complex js case, your file could have multiple functions and only one will be exported as the main one

**Reply by ChristinaHyh** [reply]
Thank you, it works!

**Reply by DannyHatcher-Git** [reply]
As someone very new to JS @ChristinaHyh could is there anything you changed about the above steps to get things working?
